nvtabular/workflow.py

```diff
@@ -13,1078 +13,178 @@
-            output_path = str(output_path)
-            self.ddf_to_dataset(
-                output_path,
-                output_format=output_format,
-                shuffle=shuffle,
-                out_files_per_proc=out_files_per_proc,
-                num_threads=num_io_threads,
-            )
-
-    def write_to_dataset(
-        self,
-        path,
-        dataset,
-        apply_ops=False,
-        out_files_per_proc=None,
-        shuffle=None,
-        output_format="parquet",
-        iterate=False,
-        nfiles=None,
-        num_io_threads=0,
-        dtypes=None,
-    ):
-        """Write data to shuffled parquet dataset.
-
-        Assumes statistics are already gathered.
-        """
-        # Check shuffle argument
-        shuffle = _check_shuffle_arg(shuffle)
-
-        if nfiles:
-            warnings.warn("nfiles is deprecated. Use out_files_per_proc")
-            if out_files_per_proc is None:
-                out_files_per_proc = nfiles
-        out_files_per_proc = out_files_per_proc or 1
 
-        path = str(path)
-        if iterate:
-            self.iterate_online(
-                dataset,
-                output_path=path,
-                shuffle=shuffle,
-                output_format=output_format,
-                out_files_per_proc=out_files_per_proc,
-                apply_ops=apply_ops,
-                num_io_threads=num_io_threads,
-                dtypes=dtypes,
-            )
-        else:
-            self.build_and_process_graph(
-                dataset,
-                output_path=path,
-                record_stats=False,
-                shuffle=shuffle,
-                output_format=output_format,
-                out_files_per_proc=out_files_per_proc,
-                apply_ops=apply_ops,
-                num_io_threads=num_io_threads,
-                dtypes=dtypes,
-            )
+def _transform_ddf(ddf, column_groups):
+    if isinstance(column_groups, ColumnGroup):
+        column_groups = [column_groups]
 
-    def ddf_to_dataset(
-        self,
-        output_path,
-        shuffle=None,
-        out_files_per_proc=None,
-        output_format="parquet",
-        num_threads=0,
-    ):
-        """Dask-based dataset output.
-
-        Currently supports parquet only.
-        """
-        if output_format not in ("parquet", "hugectr"):
-            raise ValueError("Only parquet/hugectr output supported with Dask.")
-        ddf = self.get_ddf()
-        fs = get_fs_token_paths(output_path)[0]
-        fs.mkdirs(output_path, exist_ok=True)
-        if shuffle or out_files_per_proc:
+    columns = list(flatten(cg.flattened_columns for cg in column_groups))
 
-            cat_names = self.get_final_cols_names("categorical")
-            cont_names = self.get_final_cols_names("continuous")
-            label_names = self.get_final_cols_names("label")
+    return ddf.map_partitions(
+        lambda gdf: _transform_partition(gdf, column_groups),
```

Comment:
```suggestion
        _transform_partition,
        column_groups,
```
I don't think this lambda is needed